azurerm/resource_arm_batch_account_test.go

```diff
@@ -0,0 +1,233 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func TestValidateBatchAccountName(t *testing.T) {
+	testCases := []struct {
+		input       string
+		shouldError bool
+	}{
+		{"ab", true},
+		{"ABC", true},
+		{"abc", false},
+		{"123456789012345678901234", false},
+		{"1234567890123456789012345", true},
+		{"abc12345", false},
+	}
+
+	for _, test := range testCases {
+		_, es := validateAzureRMBatchAccountName(test.input, "name")
+
+		if test.shouldError && len(es) == 0 {
+			t.Fatalf("Expected validating name %q to fail", test.input)
+		}
+	}
+}
+
+func TestAccAzureRMBatchAccount_basic(t *testing.T) {
+	resourceName := "azurerm_batch_account.test"
+	ri := acctest.RandInt()
+	rs := acctest.RandString(4)
+	rs2 := acctest.RandString(4)
+	location := testLocation()
+
+	config := testAccAzureRMBatchAccount_basic(ri, rs, rs, location)
+	configUpdate := testAccAzureRMBatchAccount_basicUpdate(ri, rs2, rs, location)
```

Comment:
we should be able to reuse either `ri` or `rs` for the name of the batch account?

Reply:
Done.